From e252d0a36e62e4be34f772f5094502ea68b768a0 Mon Sep 17 00:00:00 2001
From: Andy Whitcroft <apw@canonical.com>
Date: Wed, 30 Sep 2009 11:33:53 +0100
Subject: [PATCH] UBUNTU: updateconfigs -- handle new ARCH check in mainline

When updating configs we reuse the build directory and thereby the host
compiled build tools to generate the updated config.  This is done for
all architectures.  However mainline is now starting to check that
the directory was built for the right architecture and erroring where it
is not:

    .../Makefile:210: *** ARCH changed from "x86_64" to "i386".
                                    Use "make mrproper" to fix it up.  Stop.

This prevents updateconfigs from running correctly.

Signed-off-by: Andy Whitcroft <apw@canonical.com>
---
 debian.master/scripts/misc/kernelconfig |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/debian.master/scripts/misc/kernelconfig b/debian.master/scripts/misc/kernelconfig
index d0ef3a1..6ad903b 100755
--- a/debian.master/scripts/misc/kernelconfig
+++ b/debian.master/scripts/misc/kernelconfig
@@ -39,9 +39,10 @@ if [ "$mode" = "genconfig" ]; then
 	test -d CONFIGS || mkdir CONFIGS
 fi
 
-test -d build || mkdir build
-
 for arch in $archs; do
+	rm -rf build
+	mkdir build
+
 	# Map debian archs to kernel archs
 	case "$arch" in
 		amd64)	kernarch="x86_64"	;;
@@ -114,6 +115,7 @@ for arch in $archs; do
 done
 
 rm -f $common_conf
+rm -rf build
 
 # Now run splitconfig.pl on all the config.common.<arch> copied to
 # $tmpdir
-- 
1.6.3.rc3.199.g24398

